src/libraries/System.DirectoryServices/src/System/DirectoryServices/SearchResultCollection.cs

```diff
@@ -214,6 +214,51 @@ void ICollection.CopyTo(Array array, int index)
             InnerList.CopyTo(array, index);
         }
 
+        /// <summary>Provides an enumerator implementation for the array of <see cref="SearchResult"/>.</summary>
+        private struct AlreadyReadResultsEnumerator : IEnumerator
```

Comment:
Why struct?

Reply:
This was my suggestion; I thought it may save some allocations in some scenarios?

Reply:
It looks like the only place it's used is here:
https://github.com/dotnet/runtime/pull/113775/files#diff-c9880e3568259df335f4e47504988814365bc6213087cd341b8939595f1fd988R194
in which case it's always immediately cast to an IEnumerator, boxing it.

Reply:
We can change it back to class if it makes a difference.

Reply:
Happy to swap it back, I did look for some other uses when it was suggested.

Reply:
go ahead and change it back.

Reply:
PR #113927 submitted.